binary/relations.proto

```diff
@@ -86,6 +86,15 @@ message ReadRel {
 
       Format format = 5;
 
+      // the index of the partition this item belongs to
+      uint64 partition_index = 6;
+
+      // the start position in byte to read from this item
+      uint64 start = 7;
```

Comment:
Can you give a bit more detail what file format you are thinking of?  For example, given a parquet file, I don't think I can apply an arbitrary byte offset.  Instead I would normally expect to see a row group number of some kind.

Reply:
Thanks for your comment, and we are thinking of Parquet file. Actually to read which row group can be determined by the internal logic according the start and length. For example, in Spark, if the midpoint of a row group locates in the region [start, start + length), then this row group will be be read. In Arrow, we also added that kind of logic. Therefore, the start and length would be very useful for determining the row groups to read.

Reply:
If the start position was in the middle of a row group then would that row group be returned?

If the row group is returned should it be sliced as closely as possible to the start position?

If it is sliced what happens if the start position is in the middle of a value?  Or in the middle of a compressed buffer?

Reply:
- For Q1: In current implementation, if the midpoint of the current rowgroup is not in the range of [start, start + length), this rowgroup will not be read by this partition. But because all the partitions must include the midpoints of all the rowgroups, this rowgroup will be read by other partition, and will not be missed.
- For Q2&3: It will not be sliced, because even if it is not read by current partition, it will be read by other partition.

Reply:
Ok, that makes sense.  The reader will still read entire row groups but will only read those row groups whose midpoints are contained in the range.  Thank you for the explanation.

Reply:
Can one of you propose some additional comments and/or information in the spec docs to make sure this is clear to future users?

Reply:
I took a stab at it: #109